jabls/src/main/java/org/jabref/languageserver/ExtensionSettings.java

```diff
@@ -1,53 +1,97 @@
 package org.jabref.languageserver;
 
+import java.util.Optional;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
 
 public class ExtensionSettings {
+    private final Gson gson;
     private boolean consistencyCheck;
+    private boolean consistencyCheckRequired;
+    private boolean consistencyCheckOptional;
+    private boolean consistencyCheckUnknown;
     private boolean integrityCheck;
 
     private ExtensionSettings() {
         this.consistencyCheck = true;
+        this.consistencyCheckRequired = true;
+        this.consistencyCheckOptional = true;
+        this.consistencyCheckUnknown = true;
         this.integrityCheck = true;
+        this.gson = new GsonBuilder().create();
     }
 
     public static ExtensionSettings getDefaultSettings() {
         return new ExtensionSettings();
     }
 
-    public void setConsistencyCheck(boolean consistencyCheck) {
-        this.consistencyCheck = consistencyCheck;
+    public void copyFromJsonObject(JsonObject json) {
+        this.consistencyCheck = assignIfPresent(json, this.consistencyCheck, "jabref", "consistencyCheck", "enabled");
+        this.consistencyCheckRequired = assignIfPresent(json, this.consistencyCheckRequired, "jabref", "consistencyCheck", "required");
+        this.consistencyCheckOptional = assignIfPresent(json, this.consistencyCheckOptional, "jabref", "consistencyCheck", "optional");
+        this.consistencyCheckUnknown = assignIfPresent(json, this.consistencyCheckUnknown, "jabref", "consistencyCheck", "unknown");
+        this.integrityCheck = assignIfPresent(json, this.integrityCheck, "jabref", "integrityCheck", "enabled");
     }
 
-    public boolean isConsistencyCheck() {
-        return consistencyCheck;
+    private boolean assignIfPresent(JsonObject obj, boolean current, String... path) {
+        return assignIfPresent(obj, current, Boolean.class, path);
     }
 
-    public void setIntegrityCheck(boolean integrityCheck) {
-        this.integrityCheck = integrityCheck;
+    private <T> T assignIfPresent(JsonObject obj, T current, Class<T> type, String... path) {
+        JsonObject currentObject = obj;
+        for (String key : path) {
+            Optional<JsonElement> element = Optional.ofNullable(currentObject.get(key));
+            if (element.isEmpty()) {
+                return current;
+            }
+            if (element.get().isJsonObject()) {
+                currentObject = element.get().getAsJsonObject();
+                continue;
+            }
+            try {
+                T v = gson.fromJson(element.get(), type);
+                if (v != null) {
+                    return v;
+                }
+            } catch (JsonParseException ignore) {
```

Comment:
```suggestion
            } catch (JsonParseException _) {
```
btw, why ignore the error? maybe at least a log?